test/unified_format.py

```diff
@@ -1080,38 +1080,42 @@ def _testOperation_loop(self, spec):
         successes_key = spec.get('storeSuccessesAsEntity')
         iteration_key = spec.get('storeIterationsAsEntity')
         iteration_limiter_key = spec.get('numIterations')
-        if failure_key:
-            self.entity_map[failure_key] = []
-        if error_key:
-            self.entity_map[error_key] = []
-        if successes_key:
-            self.entity_map[successes_key] = 0
-        if iteration_key:
-            self.entity_map[iteration_key] = 0
+        for i in [failure_key, error_key]:
+            if i:
+                self.entity_map[i] = []
+        for i in [successes_key, iteration_key]:
+            if i:
+                self.entity_map[i] = 0
         i = 0
+        global IS_INTERRUPTED
         while True:
             if iteration_limiter_key and i >= iteration_limiter_key:
                 break
             i += 1
             if IS_INTERRUPTED:
                 break
             try:
+                if iteration_key:
+                    self.entity_map._entities[iteration_key] += 1
                 for op in spec["operations"]:
                     self.run_entity_operation(op)
                     if successes_key:
                         self.entity_map._entities[successes_key] += 1
-                if iteration_key:
-                    self.entity_map._entities[iteration_key] += 1
             except AssertionError as exc:
```

Comment:
Let's combine the AssertionError and Exception excepts into one:
```python
            except Exception as exc:
                if isinstance(exc, AssertionError):
                    key = failure_key or error_key
                else:
                    key = error_key or failure_key
                if not key:
                    raise
                    
                self.entity_map[key].append({
                        "error": str(exc),
                         "time": time.time(),
                         "type": type(exc)
                    })
```

Note that bare `raise` is preferred over `raise exc` since it will preserver the original traceback.

Reply:
Done. Noted.